src/Symfony/Component/Serializer/Encoder/CsvEncoder.php

```diff
@@ -109,11 +111,17 @@ public function encode($data, $format, array $context = [])
 
         if (!($context[self::NO_HEADERS_KEY] ?? $this->defaultContext[self::NO_HEADERS_KEY])) {
             fputcsv($handle, $headers, $delimiter, $enclosure, $escapeChar);
+            if ("\n" !== ($context[self::END_OF_LINE] ?? "\n") && 0 === fseek($handle, -1, \SEEK_CUR)) {
```

Comment:
my bad, this should be `($context[self::END_OF_LINE] ?? $this->defaultContext[self::END_OF_LINE] ?? "\n")`
(same below)